### PR TITLE
Make the underlying buffer of the ArrayBuffer accessible.

### PR DESCRIPTION
**Pull request by LaszloLango**
JerryScript-DCO-1.0-Signed-off-by: László Langó llango.u-szeged@partner.samsung.com

### CODE REVIEW

File: tests/unit-core/test-arraybuffer.c [hunk 1]
```diff
@@ -326,8 +326,6 @@ main (void)
       sum += data[i];
     }
 
-    jerry_release_value (buffer);
```
Comment (galpeter) [inline tests/unit-core/test-arraybuffer.c]: Why was this removed?

Comment (LaszloLango) [reply tests/unit-core/test-arraybuffer.c]: I feel this kind of implicit value acquire leads to unexpected memory leaks.

Comment (LaszloLango) [reply tests/unit-core/test-arraybuffer.c]: Furthermore it cannot stop users to try to access the pointer after the buffer is freed.

Comment (rerobika) [reply tests/unit-core/test-arraybuffer.c]: +1 for removing the extra reference. That's totally the user's responsibility to manage the buffer correctly.

Comment (galpeter) [reply tests/unit-core/test-arraybuffer.c]: ahh true

File: jerry-core/api/jerry.c [hunk 2]
```diff
@@ -3198,34 +3198,29 @@ jerry_get_arraybuffer_byte_length (const jerry_value_t value) /**< ArrayBuffer *
...
 uint8_t *
-jerry_get_arraybuffer_pointer (const jerry_value_t value) /**< Array Buffer to use */
+jerry_get_arraybuffer_pointer (const jerry_value_t array_buffer) /**< Array Buffer to use */
 {
   jerry_assert_api_available ();
+
 #if ENABLED (JERRY_ES2015_BUILTIN_TYPEDARRAY)
-  if (!ecma_is_arraybuffer (value))
+  if (ecma_is_value_error_reference (array_buffer)
+      || !ecma_is_arraybuffer (array_buffer))
```
Comment (galpeter) [inline jerry-core/api/jerry.c]: Doesn't the ecma_is_arraybuffer already checks error references?

Comment (LaszloLango) [reply jerry-core/api/jerry.c]: No it doesn't.

**Comment by galpeter** [comment]
lgtm
